lib_tflite_micro/src/tflite-xcore-kernels/xcore_lookup.cc

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022, XMOS Ltd, All rights reserved
+
+#include "MemCpyFn.hpp"
+#include "xcore_custom_options.h"
+#include "xcore_utils.h"
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace xcore {
+namespace lookup {
+
+// This is the struct that contains the data required by the operator
+struct LookupOpData
+    : XCoreOpData { // Inherits the operator name field from XCoreOpData
+};
+
+
+void *Init(TfLiteContext *context, const char *buffer, size_t length) {
+  auto op_data = construct_persistent_object<LookupOpData>(context);
+  op_data->name = "XC_lookup";
+
+  return op_data;
+}
+
+// Does all the requests for scratches
+TfLiteStatus Prepare(TfLiteContext *context, TfLiteNode *node) {
+  return kTfLiteOk;
+}
+
+TfLiteStatus Eval(TfLiteContext *context, TfLiteNode *node) {
+
+  auto *op_data = static_cast<LookupOpData *>(node->user_data);
+  
+  // Get Input/Output Tensors
+  const TfLiteEvalTensor *input = tflite::micro::GetEvalInput(context, node, 0);
+  int input_size = 1;
+  for(int i = 0; i < input->dims->size; i++) input_size *= input->dims->data[i];
+  const TfLiteEvalTensor *table = tflite::micro::GetEvalInput(context, node, 1);
+  TfLiteEvalTensor *output = tflite::micro::GetEvalOutput(context, node, 0);
+
+  // Pointers to data in In/Out Tensors
+  //void *in_data = const_cast<void *>(tflite::micro::GetTensorData<void>(input));
```

Comment:
Please remove the commented out code, thanks!

Reply:
I think::

  ((int8_t*)out_data)[i] = ((int8_t*)table_vals)[((uint8_t *)in_val)[i]]

should do it?

Reply:
Sorry - this was meant to be part of the discussion on the if() statement that can be removed. Memcpy() isn't needed either for copying one byte.

Reply:
Ok that's smart with the uint8_t cast, will use that!